### ISSUE TITLE
[ISSUE] RuntimeException for API23 and Below

### ISSUE THREAD

**Issue by davcpas1234** [VersCreativeUK/FACETS#28]
**Application Version**

0.0.6

**Issue Description**

**Reported by Google Pre Launch Report**
3 Devices tested reporting a crash upon opening the Toolkit application:

Device | Android Version (API) | Locale 
------------ | ------------- | -------------
P8 LiteHuawei | Android 5.0 (21) | English (United Kingdom)en_GB 
Moto G4 PlayMotorola | Android 6.0 (23) | English (United States)en_US 
K3 2017LGE | Android 6.0 (23) | English (United Kingdom)en_GB

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{uk.ac.bournemouth.facets/uk.ac.bournemouth.facets.Activities.SplashActivity}: android.view.InflateException: Binary XML file line #21: Binary XML file line #21: Error inflating class androidx.appcompat.widget.AppCompatImageView
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2429)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2493)
	at android.app.ActivityThread.-wrap11(ActivityThread.java)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1357)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:148)
	at android.app.ActivityThread.main(ActivityThread.java:5459)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:728)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:618)
Caused by: android.view.InflateException: Binary XML file line #21: Binary XML file line #21: Error inflating class androidx.appcompat.widget.AppCompatImageView
	at android.view.LayoutInflater.inflate(LayoutInflater.java:539)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:423)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:374)
	at androidx.appcompat.app.AppCompatDelegateImpl.setContentView(AppCompatDelegateImpl.java:469)
	at androidx.appcompat.app.AppCompatActivity.setContentView(AppCompatActivity.java:140)
	at uk.ac.bournemouth.facets.Activities.SplashActivity.onCreate(SplashActivity.java:22)
	at android.app.Activity.performCreate(Activity.java:6259)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1130)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2382)
	... 9 more
Caused by: android.view.InflateException: Binary XML file line #21: Error inflating class androidx.appcompat.widget.AppCompatImageView
	at android.view.LayoutInflater.createView(LayoutInflater.java:645)
	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:764)
	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:704)
	at android.view.LayoutInflater.rInflate(LayoutInflater.java:835)
	at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:798)
	at android.view.LayoutInflater.rInflate(LayoutInflater.java:838)
	at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:798)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:515)
	... 17 more
Caused by: java.lang.reflect.InvocationTargetException
	at java.lang.reflect.Constructor.newInstance(Native Method)
	at android.view.LayoutInflater.createView(LayoutInflater.java:619)
	... 24 more
Caused by: android.content.res.Resources$NotFoundException: Resource ID #0x7f0800b9
	at android.content.res.Resources.getValue(Resources.java:1351)
	at androidx.appcompat.widget.AppCompatDrawableManager.loadDrawableFromDelegates(AppCompatDrawableManager.java:331)
	at androidx.appcompat.widget.AppCompatDrawableManager.getDrawable(AppCompatDrawableManager.java:198)
	at androidx.appcompat.widget.AppCompatDrawableManager.getDrawable(AppCompatDrawableManager.java:191)
	at androidx.appcompat.content.res.AppCompatResources.getDrawable(AppCompatResources.java:102)
	at androidx.appcompat.widget.AppCompatImageHelper.loadFromAttributes(AppCompatImageHelper.java:59)
	at androidx.appcompat.widget.AppCompatImageView.<init>(AppCompatImageView.java:78)
	at androidx.appcompat.widget.AppCompatImageView.<init>(AppCompatImageView.java:68)
	... 26 more
```

**Steps to Reproduce**

1. Open the App
2. Crash

**Expected Behaviour**

Opens as normal

**Screenshots (Optional)**

![image](https://user-images.githubusercontent.com/3764587/52351372-87f8e080-2a22-11e9-9255-1378c0bf02b2.png)

**Comment by davcpas1234** [comment]
This is caused by an incompatibility issue with the BU logo on the Splash screen:

```
<gradient android:endX="453.793" android:endY="165.6182" android:startX="237.2373" android:startY="165.6182" android:type="linear">
    <item android:color="#FFF26F39" android:offset="0.005618"/>
    <item android:color="#FFE9316B" android:offset="1"/>
</gradient>
```

`endX`, `endY`, `startX`, `startY` and `offset` functions were only introduced in API 24, therefore this resource is stored purely in `drawables-v24` and there is no equivalent in `drawables`.

This is the issue as there is "technically" no resource `ic_bournemouth_university` for v23 and below

**Comment by davcpas1234** [comment]
There will now be 2 versions of the logo depending on a device's API level.

API 21-23 will see this:

![image](https://user-images.githubusercontent.com/3764587/52353964-76660780-2a27-11e9-8a86-6e000e56b2c4.png)

and API 24 and above will see this:

![image](https://user-images.githubusercontent.com/3764587/52354016-8da4f500-2a27-11e9-8213-0b7e0a63f268.png)

**Comment by davcpas1234** [comment]
Released in `0.0.6-rc3`

**Comment by davcpas1234** [comment]
Reopening this issue, due to the API 24 version not being displayed and the "fallback" always being displayed.

After multiple iterations, no solution has been found. As a result I have reached out to the development community for support:
https://stackoverflow.com/questions/54882710/vector-drawable-not-overriding

Will update

**Comment by davcpas1234** [comment]
Thanks to the support of the dev community the below fix has been implemented:

```
if(Build.VERSION.SDK_INT <= Build.VERSION_CODES.M){
    logo.setImageDrawable(ResourcesCompat.getDrawable(getResources(), R.drawable.ic_bournemouth_university, null));
}else{
    logo.setImageDrawable(ResourcesCompat.getDrawable(getResources(), R.drawable.ic_bournemouth_university_24, null));
}
````

**Comment by davcpas1234** [comment]
Workaround will be released in `0.0.7-rc2`
